File: src/types/market.ts
export interface StockData {
  symbol: string
  price: number
  change: number
}

export interface MarketOverview {
  stocks: StockData[]
  market_status: 'OPEN' | 'CLOSED'
  as_of: string
}

export interface SectorData {
  sector: string
  change: number
}

export interface InstitutionalNews {
  investor_name: string
  title: string
  url: string
  date: string
}

export interface FiiDiiFlow {
  date: string
  fii: number
  dii: number
}

export interface BulkDeal {
  ticker: string
  client: string
  qty: number
  price: number
  type: string
  date: string
}

export interface MarketEvent {
  id: number
  ticker: string
  type: string
  description: string
  ts: string
  status: string
}

File: backend/schemas/market.py
from pydantic import BaseModel, Field


class StockData(BaseModel):
    symbol: str
    price: float
    change: float


class MarketOverview(BaseModel):
    stocks: list[StockData] = Field(default_factory=list)
    market_status: str
    as_of: str


class SectorPerformance(BaseModel):
    sector: str
    change: float


class FiiDiiFlow(BaseModel):
    date: str
    fii: float
    dii: float


class BulkDeal(BaseModel):
    ticker: str
    client: str
    qty: float
    price: float
    type: str
    date: str


class MarketEvent(BaseModel):
    id: int
    ticker: str
    type: str
    description: str
    ts: str
    status: str


class InvestorNews(BaseModel):
    investor_name: str
    title: str
    url: str
    date: str
